src/Runtime/OMTensorList.inc

```diff
@@ -73,40 +73,26 @@ struct OMTensorList {
 /* OMTensorList creator */
 OMTensorList *omTensorListCreate(OMTensor **tensors, int64_t n) {
   OMTensorList *list = (OMTensorList *)malloc(sizeof(struct OMTensorList));
-  if (!list)
-    return NULL;
-  list->_size = n;
-  list->_omts = (OMTensor **)malloc(sizeof(OMTensor *) * n);
-  list->_shallow = false;
-
-  if (!list->_omts) {
-    free(list); // free the previously allocated memory in case of an error
-    return NULL;
-  }
-  // copy the pointers to the OMTensor array
-  // now OMTensorList will always own the tensor array of pointers
-  memcpy(tensors, list->_omts, n);
-
-  return list;
+  omTensorListCreateWithShallow(tensors, n, false);
```

Comment:
I think this leaks the `list` and is missing the return statement.

```
OMTensorList *omTensorListCreate(OMTensor **tensors, int64_t n) {
  return omTensorListCreateWithShallow(tensors, n, false);
}
```

Reply:
Fixed